core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerProxy.scala

```diff
@@ -581,19 +581,30 @@ class ContainerProxy(
               val initRunInterval = initInterval
                 .map(i => Interval(runInterval.start.minusMillis(i.duration.toMillis), runInterval.end))
                 .getOrElse(runInterval)
-              ContainerProxy.constructWhiskActivation(job, initInterval, initRunInterval, response)
+              ContainerProxy.constructWhiskActivation(
+                job,
+                initInterval,
+                initRunInterval,
+                runInterval.duration >= actionTimeout,
```

Comment:
should this be strictly `>`? the action completed here and returned a response.

Reply:
We tried this, but then the test `KubernetesContainerTests > KubernetesContainer should properly deal with a timeout during run` failed. The testcase using nano seconds whereas the core code uses milli seconds as unit. We talked with @sven-lange-last and decided to revert back to the old implementation based on milli seconds. Sven suggested to not simply change the testcase and keep the lower than comparison for timeout to fix the problem since this change could imply other side effects which should be handled in a separate issue. This issue should also cover changes to use nano seconds for duration instead of milli seconds. 

```
org.apache.openwhisk.core.containerpool.kubernetes.test.KubernetesContainerTests > KubernetesContainer should properly deal with a timeout during run FAILED
    org.scalatest.exceptions.TestFailedException: (Interval(1970-01-01T00:00:00Z,1970-01-01T00:00:01Z),{"result":{"error":"The action did not produce a valid response and exited unexpectedly."},"statusCode":2}) was not equal to (Interval(1970-01-01T00:00:00Z,1970-01-01T00:00:01Z),{"result":{"error":"The action exceeded its time limits of 1000 milliseconds."},"statusCode":2})
```

Reply:
thank you for the thoughtful explanation.

Reply:
additional remark to the comment above regarding "old implementation based on milli seconds": it is meant that we have changed back to `<=` comparisons with timeouts.